cli/dcoscli/package/main.py

```diff
@@ -517,3 +525,205 @@ def _uninstall(package_name, remove_all, app_id, cli, app):
         return 1
 
     return 0
+
+
+def _bundle(package_directory, output_directory):
+    """
+    :param package_directory: directory containing the package
+    :type package_directory: str
+    :param output_directory: directory where to save the package zip file
+    :type output_directory: str
+    :returns: process status
+    :rtype: int
+    """
+
+    if output_directory is None:
+        output_directory = os.getcwd()
+    logger.debug('Using [%s] as the ouput directory', output_directory)
+
+    # Find package.json file and parse it
+    package_json = _validate_json_file(
+        os.path.join(package_directory, 'package.json'))
+
+    with tempfile.NamedTemporaryFile() as temp_file:
+        with zipfile.ZipFile(
+                temp_file.name,
+                mode='w',
+                compression=zipfile.ZIP_DEFLATED,
+                allowZip64=True) as zip_file:
+            # list through package directory and add files zip archive
+            for filename in sorted(os.listdir(package_directory)):
+                fullpath = os.path.join(package_directory, filename)
+                if filename == 'marathon.json.mustache':
+                    zip_file.write(fullpath, arcname=filename)
+                elif (filename == 'config.json' or
+                        filename == 'command.json' or
+                        filename == 'package.json'):
+                    # schema check the config and command json file
+                    _validate_json_file(fullpath)
+                    zip_file.write(fullpath, arcname=filename)
+                elif filename == 'assets' and os.path.isdir(fullpath):
+                    _import_assets(fullpath, zip_file)
+                elif filename == 'images' and os.path.isdir(fullpath):
+                    _import_images(fullpath, zip_file)
+                else:
+                    # anything else is an error
+                    raise DCOSException('Unknown file [{}]'.format(fullpath))
+
+        # Compute the name of the package file
+        zip_file_name = os.path.join(
+            output_directory,
+            '{}-{}-{}.zip'.format(
+                package_json['name'],
+                package_json['version'],
+                _hashfile(temp_file.name)))
+
+        if os.path.exists(zip_file_name):
+            raise DCOSException(
+                'Output file [{}] already exists'.format(
+                    zip_file_name))
+
+        # rename with digest
+        util.sh_copy(temp_file.name, zip_file_name)
+
+    # Print the full path to the file
+    emitter.publish(
+        errors.DefaultError(
+            'Created DCOS Universe package [{}].'.format(zip_file_name)))
```

Comment:
For the DCOS CLI, I want to keep all text intended for human in stderr and messages intended for scripts, etc in stdin.